cli/pkg/flag/kind.go

```diff
@@ -0,0 +1,35 @@
+// Copyright © 2020 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package flag
+
+import "fmt"
+
+// Kind describe kind of Tekton Resource
+type Kind string
+
+// Kinds of Tekton Resources
+const (
+	Task     Kind = "task"
+	Pipeline      = "pipeline"
+)
+
+// IsValid validates if a resource kind is valid Tekton kind
+func (r Kind) IsValid() error {
+	switch r {
+	case Task, Pipeline:
+		return nil
+	}
+	return fmt.Errorf("invalid value %q set for option kinds. Valid options: [task, pipeline]", r)
+}
```

Comment:
Any reason to not use `tektoncd/pipeline` types ? (except not depending on it, which… make sense to me :wink: )

Reply:
+1 currently we don't need to depend on tektoncd/pipelines but in future we might have to esp.. if we want to support `tkn hub install task buildah` .

Reply:
@vdemeester is there a set of consts/enums that tektoncd/pipelines define we can reuse?